src/backend/distributed/transaction/remote_transaction.c

```diff
@@ -865,11 +865,12 @@ ResetRemoteTransaction(struct MultiConnection *connection)
 	RemoteTransaction *transaction = &connection->remoteTransaction;
 
 	/* unlink from list of open transactions, if necessary */
-	if (transaction->transactionState != REMOTE_TRANS_NOT_STARTED)
+	if (connection->transactionNode.next)
```

Comment:
Would be nice to have an explicit boolean variable for this

Reply:
On one hand I agree. On the other hand I like that with this approach you don't have to keep two variables in sync.

Reply:
Sort of. You still have to memset yourself, so with the current approach you only save some bookkeeping on the push side, which only happens in one place.